recipes/cspice/all/conanfile.py

```diff
@@ -0,0 +1,160 @@
+import os
+import time
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class CspiceConan(ConanFile):
+    name = "cspice"
+    description = "NASA C SPICE library"
+    license = "TSPA"
+    topics = ("conan", "spice", "naif", "kernels", "space", "nasa", "jpl", "spacecraft", "planet", "robotics")
+    homepage = "https://naif.jpl.nasa.gov/naif/toolkit.html"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    exports = ["TSPA.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _patch_per_platform(self):
+        return {
+            "Macos": {
+                "x86": {
+                    "apple-clang": ["to-MacIntel-OSX-or-SunSPARC-Solaris.patch", "to-MacIntel-OSX-AppleC-32bit.patch"]
+                },
+                "x86_64": {
+                    "apple-clang": ["to-MacIntel-OSX-or-SunSPARC-Solaris.patch", "to-MacIntel-OSX-AppleC-64bit.patch"]
+                }
+            },
+            "Linux": {
+                "x86": {
+                    "gcc": ["to-PC-Linux-GCC-32bit.patch"]
+                },
+                "x86_64": {
+                    "gcc": ["to-PC-Linux-GCC-64bit.patch"]
+                }
+            },
+            "Windows": {
+                "x86": {
+                    "Visual Studio": ["to-PC-Windows.patch", "to-PC-Windows-VisualC-32bit.patch"]
+                },
+                "x86_64": {
+                    "Visual Studio": ["to-PC-Windows.patch", "to-PC-Windows-VisualC-64bit.patch"]
+                }
+            },
+            "cygwin": {
```

Comment:
We don't classify cygwin as OS. You should move it to Windows context:

 "Windows": { "x86": { "gcc": [] } }

Reply:
But it is supported as a subystem of windows. I use this function:
```python
def _get_os_or_subsystem(self):
    if self.settings.os == "Windows" and self.settings.os.subsystem != "None":
        os_or_subsystem = str(self.settings.os.subsystem)
    else:
        os_or_subsystem = str(self.settings.os)
    return os_or_subsystem
```

Reply:
Thanks for clarifying!